### PR TITLE
[JDG-2062] - Export Data Grid stats to Prometheus

### PR DESCRIPTION
**Pull request by vblagoje**
Initial implementation of JDG-2062. All DG stats currently available through JMX are to be exported to Prometheus. Needs a rebase.

### CODE REVIEW

File: modules/datagrid/launch/added/launch/prometheus.conf [hunk 1]
```diff
@@ -19,6 +19,9 @@
 # Prometheus agent JAR path in our image
 PROMETHEUS_JAR=$(getfiles io/prometheus/jmx/jdg-7.3/)
 
+# Create a symlink to PROMETHEUS_JAR so it could be referenced easily in case JAVA_OPTS_APPEND env is used to turn on Prometheus
+ln -s ${PROMETHEUS_JAR} ${JBOSS_HOME}/jmx_prometheus_javaagent.jar
```
Comment (ryanemerson) [inline modules/datagrid/launch/added/launch/prometheus.conf:18]: Nitpick, but if this is for convenience why not just call it `prometheus.jar`

Comment (Crumby) [reply modules/datagrid/launch/added/launch/prometheus.conf:18]: `jmx_exporter.jar` would be most accurate.

Comment (ryanemerson) [reply modules/datagrid/launch/added/launch/prometheus.conf:18]: `jmx_prometheus_javaagent-0.3.1.redhat-00001.jar` would be the most accurate :stuck_out_tongue_winking_eye:

I'm happy with anything that abstracts away the version number.

File: modules/datagrid/launch/added/launch/prometheus.conf [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+# Configure JVM options for Prometheus Agent
+#
+# Env Vars respected:
+#
+# AB_PROMETHEUS_ENABLE: Enable the use of the Prometheus agent.
+#
+# AB_PROMETHEUS_JMX_EXPORTER_PORT: Port to use for the Prometheus JMX Exporter.
+#   Defaults to 9779
+#
+# AB_PROMETHEUS_JMX_EXPORTER_CONFIG: Path to configuration to use for the
```
Comment (Crumby) [inline modules/datagrid/launch/added/launch/prometheus.conf]: This help is misleading. Exporter config variable is no longer present.

**Review by osnipso — APPROVED** [review]
LGTM. Despite CI having failed, it was a timeout. I ran this PR code in my local box, got `Tests run: 37, Failures: 0, Errors: 0, Skipped: 1`. The skipped test is from CachingServiceTest, we might take a look later.
I'm merging this PR.

**Pull request merged**